// Copyright (c) 2021 Huawei Device Co., Ltd.
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
option java_package = "ohos.devtools.datasources.transport.grpc.service";
option optimize_for = LITE_RUNTIME;

message HilogDetails {
    // log time
    uint64 tv_sec = 1;
    uint64 tv_nsec = 2;
    uint32 pid = 3;
    uint32 tid = 4;
    uint32 level = 5;
    string tag = 6;
}

message HilogLine {
    HilogDetails detail = 1;
    string context = 2;
    uint64 id = 3;
}

message HilogInfo {
    repeated HilogLine info = 1;
    uint32 clock = 2; // 空值
}